app/scripts/controllers/network/network-controller.js

```diff
@@ -22,14 +23,23 @@ import {
   CHAIN_IDS,
   NETWORK_TYPES,
 } from '../../../../shared/constants/network';
+import getFetchWithTimeout from '../../../../shared/modules/fetch-with-timeout';
 import {
   isPrefixedFormattedHexString,
   isSafeChainId,
 } from '../../../../shared/modules/network.utils';
-import getFetchWithTimeout from '../../../../shared/modules/fetch-with-timeout';
 import createInfuraClient from './createInfuraClient';
 import createJsonRpcClient from './createJsonRpcClient';
 
+/**
+ * @typedef {object} NetworkConfiguration
+ * @property {string} rpcUrl - RPC target URL.
+ * @property {string} chainId - Network ID as per EIP-155
+ * @property {string} ticker - Currency ticker.
+ * @property {object} rpcPrefs - Personalized preferences.
+ * @property {string} [chainName] - Personalized network name.
```

Comment:
As just mentioned on our last meeting, leaving this as `nickname` might make bringing these changes to core/mobile much easier.

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/75241de5a91a5bdc30492f8b1865badd93f699bd